src/category_theory/bicategory/functor.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2022 Yuma Mizuno. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuma Mizuno
+-/
+import category_theory.bicategory.basic
+
+/-!
+# Oplax functors
+
+An oplax functor `F` between bicategories `B` and `C` consists of
+* a function between objects `F.obj : B ⟶ C`,
+* a family of functions between 1-morphisms `F.map : (a ⟶ b) → (obj a ⟶ obj b)`,
+* a family of functions between 2-morphisms `F.map₂ : (f ⟶ g) → (map f ⟶ map g)`,
+* a family of 2-morphisms `F.map_id a : F.map (𝟙 a) ⟶ 𝟙 (F.obj a)`,
+* a family of 2-morphisms `F.map_comp f g : F.map (f ≫ g) ⟶ F.map f ≫ F.map g`, and
+* certain consistency conditions on them.
+
+## Main definitions
+
+* `oplax_functor B C` : an oplax functor between bicategories `B` and `C`
+* `oplax_functor.comp F G` : the composition of oplax functors
+
+## Future work
+
+There are two types of functors between bicategories, called lax and oplax functors, depending on
+the directions of `map_id` and `map_comp`. We may need both in mathlib in the future, but for
+now we only define oplax functors.
+
+In addition, if we require `map_id` and `map_comp` to be isomorphisms, we obtain the definition
+of pseudofunctors. There are several possible design choices to implement pseudofunctors,
+but the choice is left to future development.
+-/
+
+set_option old_structure_cmd true
+
+namespace category_theory
+
+open category bicategory
+open_locale bicategory
+
+universes w₁ w₂ w₃ v₁ v₂ v₃ u₁ u₂ u₃
+
+section
+
+/--
+A prelax functor between bicategories consists of functions between objects,
+1-morphisms, and 2-morphisms. This structure will be extended to define `oplax_functor`.
+-/
+structure prelax_functor
+  (B : Type u₁) [quiver.{v₁+1} B] [∀ a b : B, quiver.{w₁+1} (a ⟶ b)]
+  (C : Type u₂) [quiver.{v₂+1} C] [∀ a b : C, quiver.{w₂+1} (a ⟶ b)]
+  extends prefunctor B C : Type (max w₁ w₂ v₁ v₂ u₁ u₂) :=
+(map₂ {a b : B} {f g : a ⟶ b} : (f ⟶ g) → (map f ⟶ map g))
+
+/-- The prefunctor between the underlying quivers. -/
+add_decl_doc prelax_functor.to_prefunctor
+
+variables {B : Type u₁} [quiver.{v₁+1} B] [∀ a b : B, quiver.{w₁+1} (a ⟶ b)]
+variables {C : Type u₂} [quiver.{v₂+1} C] [∀ a b : C, quiver.{w₂+1} (a ⟶ b)]
+variables (F : prelax_functor B C)
+
+@[simp] lemma prelax_functor.to_prefunctor_obj : F.to_prefunctor.obj = F.obj := rfl
```

Comment:
These lemmas set, for example, `F.map` to be simpler than `F.to_prefunctor.map`. Since `F.to_prelax_functor.to_prefunctor.map` is too long to read during the proof, I wanted to display it as `F.map`. Is this a bad idea?

Reply:
I now see that `F.to_prefunctor.map` is different from `F.map` here, because you enabled `old_structure_cmd`: https://leanprover-community.github.io/archive/stream/113488-general/topic/old_structure_cmd.html So here the simp lemmas does make sense.
If it's set to false, then `F.map` is just an alias of `F.to_prefunctor.map`, and the simp lemma would not make sense (tactic display would always show the latter).